ydb/docs/en/core/reference/ydb-cli/tools-infer.md

```diff
@@ -0,0 +1,159 @@
+# Table schema inference from data files
+
+You can use the `{{ ydb-cli }} tools infer csv` command to generate a `CREATE TABLE` statement from a CSV data file. This can be very helpful when you want to [import](./export-import/import-file.md) data into a database and the table has not been created yet.
+
+Command syntax:
+
+```bash
+{{ ydb-cli }} [global options...] tools infer csv [options...] <input files...>
+```
+
+- `global options` – [global options](commands/global-options.md).
+- `options` – [subcommand options](#options).
+
+To get the most up-to-date information about the command, use the `--help` option:
+
+```bash
+{{ ydb-cli }} tools infer csv --help
+```
+
+## Subcommand options {#options}
+
+Option Name | Description
+---|---
+`-p, --path` | Database path to table that should be created. Default: `table`.
+`--columns` | Explicitly specifies table column names, as a comma-separated list.
+`--gen-columns` | Explicitly indicates that table column names should be generated automatically (column1, column2, ...).
+`--header` | Explicitly indicates that the first row in the CSV contains column names.
+`--rowsalyze` | Number of rows to analyze. 0 means unlimited. Reading will be stopped soon after this number of rows is read. Default: `500000`.
```

Comment:
Why is the option name so odd? It sounds like it should be like `--rows-to-analyze`.